core/trino-main/src/main/java/io/trino/sql/planner/optimizations/AddLocalExchanges.java

```diff
@@ -261,9 +261,13 @@ public PlanWithProperties visitTableFinish(TableFinishNode node, StreamPreferred
         public PlanWithProperties visitTopN(TopNNode node, StreamPreferredProperties parentPreferences)
         {
             if (node.getStep() == TopNNode.Step.PARTIAL) {
+                StreamPreferredProperties requiredProperties = parentPreferences.withoutPreference().withDefaultParallelism(session);
+                if (node.isInputPresorted()) {
+                    requiredProperties = parentPreferences.withoutPreference().withOrderSensitivity();
```

Comment:
Can we improve `StreamPropertyDerivations.Visitor#visitExchange` instead to preserve order property when single input stream is distributed with local exchange?

Reply:
https://github.com/trinodb/trino/pull/7589